telegram-bot/src/main/kotlin/com/ajaxproject/telegrambot/bot/FinanceBot.kt

```diff
@@ -7,43 +7,47 @@ import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Component
 import org.telegram.telegrambots.bots.TelegramLongPollingBot
 import org.telegram.telegrambots.meta.api.objects.Update
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
 
 @Component
 class FinanceBot(
-    val telegramUpdateDispatcher: TelegramUpdateDispatcher,
-    val userSessionService: UserSessionService,
-    val botProperties: BotProperties,
+    private val telegramUpdateDispatcher: TelegramUpdateDispatcher,
+    private val userSessionService: UserSessionService,
+    private val botProperties: BotProperties,
 ) : TelegramLongPollingBot(botProperties.token) {
 
     override fun getBotUsername(): String = botProperties.username
 
     override fun onUpdateReceived(update: Update) {
-        if ((!update.hasMessage() || !update.message.hasText()) && !update.hasCallbackQuery()) {
-            log.warn("Unexpected update from user")
-            return
-        }
+        Mono.just(update)
+            .filter { update.hasMessage() || update.hasCallbackQuery() }
+            .map { toUpdateRequest(it) }
+            .map { telegramUpdateDispatcher.dispatch(it) }
+            .doOnNext { isDispatched -> logNotDispatched(isDispatched, update) }
+            .subscribeOn(Schedulers.boundedElastic())
+            .subscribe()
+    }
 
-        val chatId = when {
-            update.message != null -> update.message.chatId
-            update.callbackQuery != null -> update.callbackQuery.message.chatId
-            else -> return
+    private fun logNotDispatched(isDispatched: Boolean, update: Update) {
+        val userId: Long = (update.message?.from?.id ?: update.callbackQuery?.message?.from?.id) as Long
+        val textFromUser: String = (update.message?.text ?: update.callbackQuery?.message?.text).toString()
+
+        if (isDispatched) {
```

Comment:
in my opinion - strange logs on the input parameters of the method. In general, this parameter is no longer used in the method. Maybe you can refuse it?